### PR TITLE
A bit faster casper tests and CI

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
bors try

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperTestUtil.scala [hunk 1]
```diff
@@ -100,11 +100,12 @@ object MultiParentCasperTestUtil {
...
     (for {
+      //TODO consider reverting back to the production-used variant
       activeRuntime <- Runtime
                         .createWithEmptyCost[Task, Task.Par](
-                          storageDirectory,
-                          storageSize,
-                          StoreType.LMDB
+                          Paths.get("/not/a/path"),
+                          -1,
+                          StoreType.InMem
```
Comment (adaszko) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperTestUtil.scala]: Is InMem this something like a Java HashTable?  If so, I think it is risky to use that.  I'd rather try to disable `fsync` in LMDB.

Comment (adaszko) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperTestUtil.scala]: See `metasync` and `sync` options at https://lmdb.readthedocs.io/en/release/#environment-class

Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperTestUtil.scala]: WDYM by risky? Do you expect more errors? Or more false-positives?

If the latter, I'd say having 200 tests not targeted at the false-positives exercise the tupplespace is an overkill and a waste.

Re concurrency semantics - @dzajkowski WDYT?

Comment (adaszko) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperTestUtil.scala]: > WDYM by risky? Do you expect more errors? Or more false-positives?

I mean any difference in behavior compared to actual LMDB.  Take for instance the long standing bug of segfaulting at shutdown.  Will it happen with InMem?  I think it should and I bet it won't.

**Comment by ArturGajowy** [comment]
Let's try this without the InMem change and see how it rolls :)

**Comment by ArturGajowy** [comment]
bors try

**Comment by ArturGajowy** [comment]
> Let's try this without the InMem change and see how it rolls :)

Yup, no visible difference. Let's use the production setting then.

**Pull request merged**